components/base/kinematicbase/ptgKinematics.go

```diff
@@ -0,0 +1,127 @@
+// Package kinematicbase contains wrappers that augment bases with information needed for higher level
+// control over the base
+package kinematicbase
+
+import (
+	"context"
+	"errors"
+	"math"
+	"time"
+
+	"github.com/golang/geo/r3"
+	"go.uber.org/multierr"
+	utils "go.viam.com/utils"
+
+	"go.viam.com/rdk/components/base"
+	"go.viam.com/rdk/motionplan/tpspace"
+	"go.viam.com/rdk/referenceframe"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+// Define a default speed to target for the base in the case where one is not provided.
+const defaultBaseMMps = 600.
+
+const (
+	ptgIndex int = iota
+	trajectoryIndexWithinPTG
+	distanceAlongTrajectoryIndex
+)
+
+type ptgBaseKinematics struct {
+	base.Base
+	frame referenceframe.Frame
+	fs    referenceframe.FrameSystem
+	ptgs  []tpspace.PTG
+}
+
+// wrapWithPTGKinematics takes a Base component and adds a PTG kinematic model so that it can be controlled.
+func wrapWithPTGKinematics(
+	ctx context.Context,
+	b base.Base,
+) (KinematicBase, error) {
+	properties, err := b.Properties(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	if properties.TurningRadiusMeters <= 0 {
+		return nil, errors.New("can only wrap with PTG kinematics if Base property TurningRadiusMeters is greater than zero")
+	}
+
+	baseMillimetersPerSecond := defaultBaseMMps // Currently no way to get this out of properties
+
+	geometries, err := b.Geometries(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	frame, err := referenceframe.NewPTGFrameFromTurningRadius(
+		b.Name().ShortName(),
+		baseMillimetersPerSecond,
+		properties.TurningRadiusMeters,
+		0,
+		geometries,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	fs := referenceframe.NewEmptyFrameSystem("")
+	if err := fs.AddFrame(frame, fs.World()); err != nil {
+		return nil, err
+	}
+
+	ptgProv, ok := frame.(tpspace.PTGProvider)
```

Comment:
This should be unnecessary if we are returning the PTGFrame and we no longer need to cast right?